iati/core/rulesets.py

```diff
@@ -126,52 +126,63 @@ def __init__(self, xpath_base, case):
 
         """
         self.case = case
-        self.xpath_base = self._valid_xpath_base(xpath_base)
+        self.context = self._valid_context(context)
         self._valid_rule_configuration(case)
         self._set_case_attributes(case)
         self._normalize_xpaths()
 
-    def _valid_xpath_base(self, xpath_base):
-        """Check that a valid `xpath_base` is given for a Rule.
+    def _valid_context(self, context):
+        """Check that a valid `context` is given for a Rule.
 
         Args:
-            xpath_base(str): The root of an XPath query.
+            context (str): The XPath expression that selects XML elements that the Rule acts against.
 
         Returns:
-            str: A valid XPath root.
+            str: A valid XPath.
 
         Raises:
             TypeError: When an argument is given that is not a string.
 
         """
-        if isinstance(xpath_base, six.string_types):
-            return xpath_base
+        if isinstance(context, six.string_types):
+            if context != '':
+                return context
+            raise ValueError
         raise TypeError
 
     def _normalize_xpath(self, path):
-        """Normalize a single XPath by combining it with `xpath_base`.
+        """Normalize a single XPath by combining it with `context`.
 
         Args:
-            path(str): An XPath.
+            path (str): An XPath.
 
         Raises:
-            AttributeError: When the `xpath_base` isn't set.
+            AttributeError: When the `context` isn't set.
 
         Todo:
             Add some logging.
+            Re-evaluate this.
 
         """
         if path == '':
-            return self.xpath_base
-        return '/'.join([self.xpath_base, path])
+            raise ValueError
+        return '/'.join([self.context, path])
 
     def _normalize_xpaths(self):
-        """Normalize xpaths by combining them with `xpath_base`.
+        """Normalize xpaths by combining them with `context`.
 
         May be overridden in child class that does not use `paths`.
```

Comment:
Should probably be a `Warning` or `Note` along with something about `_normalize_condition`

Reply:
done